clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -6474,14 +7078,17 @@ run_lockstep_simulation_for_kernel_seq(scheduler_t &scheduler, int num_outputs,
             }
             assert(status == scheduler_t::STATUS_OK);
             // Ensure stream API and the trace records are consistent.
-            assert(outputs[i]->get_input_interface()->get_tid() == IDLE_THREAD_ID ||
+            assert(outputs[i]->get_input_interface() == nullptr ||
```

Comment:
When is it null? During synthetic kernel injection? That isn't documented in scheduler.h: add a note that it can return nullptr.

Reply:
A special input sounds nicer to me than this nullptr case: the nullptr sounds like a hassle that is going to break existing code. The special input might confuse context switch checking; but if it uses the same tid,pid and workload ordinal that should solve that.

Reply:
> The special input might confuse context switch checking

Shouldn't context switch checking use `output_stream->get_tid()` instead of `output_stream->get_input_interface()->get_tid()`?

Reply:
And this special input would return the same tid as the original one? What about other APIs like get_record_ordinal and get_instruction_ordinal. They could return the ordinal within the sequence?